src/ixdat/spectra.py

```diff
@@ -508,6 +508,26 @@ def from_spectrum_list(cls, spectrum_list, **kwargs):
         obj_as_dict.update(kwargs)
         return cls.from_dict(obj_as_dict)
 
+    @property
+    def field(self):
+        """Since a spectrum can be loaded lazily, we make sure the field is loaded
+
+        We also want to make sure that the field has the tstamp of the SpectrumSeries.
+        """
+        if isinstance(self._field, PlaceHolderObject):
+            self._field = self._field.get_object()
+        if not self._field.axes_series[0].tstamp == self.tstamp:
+            self._field = Field(
+                name=self._field.name,
+                data=self._field.data,
+                unit_name=self._field.unit_name,
+                axes_series=[
+                    time_shifted(self._field.axes_series[0], tstamp=self.tstamp),
+                    self._field.axes_series[1],
+                ],
+            )
+        return self._field
```

Comment:
I think there is a case here when this method could return `None` when `self._field` is not a `PlaceHolderObject` and when the first axis has different timestamp than the one in the object.
Also, if `tstamp`s are `floats` they shouldn't be compared with `==`.

Reply:
Nice catches! 

I guess that conditional should instead be 
`if np.abs(self._field.axes_series[0].tstamp - self.tstamp) > tolerance:`
But I don't know what `tolerance` should be? It could be widely different depending on the method. Maybe
```
t = self._field.t    # shortcut to self._field.axes_series[0].data
tolerance = 1/100 * (t[-1] - t[0]) / len(t)`
```
?

I think this will never return `None`. `self._field` can only be a `Field` or a `PlaceHolderObject`. If it is a `Field` which already has the same `tstamp` in its axes series as the `Measurement`s tstamp, then it can be returned without modification.

Reply:
I've implemented the above solution, calling it `t_tolerance`. Let me know what you think!

Reply:
Maybe [numpy.is_close()](https://numpy.org/doc/stable/reference/generated/numpy.isclose.html) would be sufficient?

Saying that something "will never happen" in programming means that it is going to happen in a very weird edge case or while adding a new feature. :smile: Now it looks it will always contain at least a `PlaceHolderObject`, so we can have a look later.